### PR TITLE
Add tests for genotypes api, fix contig ordering

### LINKED ISSUE #636: Test the cycledash.genotypes module

### PR DESCRIPTION
**Pull request by ihodes**
fix https://github.com/hammerlab/cycledash/issues/407 & adds regression test for it + lots of other genotypes tests (ticking off some todos in #636, and setting up easy template for the rest)

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/cycledash/678)

### CODE REVIEW

File: common/helpers.py [hunk 1]
```diff
@@ -66,6 +66,11 @@ def find(iterable, pred):
             return x
 
 
+def pluck(obj, *keys):
```
Comment (tavinathanson) [inline common/helpers.py]: http://funcy.readthedocs.org/en/latest/colls.html#pluck?

File: cycledash/genotypes.py [hunk 2]
```diff
@@ -92,16 +92,16 @@ def get(vcf_id, query, with_stats=True):
      limit: 250
     }
     """
-    query = _annotate_query_with_types(query, spec(vcf_id))
-    compare_to_vcf_id = query.get('compareToVcfId')
+    query = _annotate_query_with_types(query, spec(run_id))
+    compare_to_run_id = query.get('compareToVcfId')
```
Comment (tavinathanson) [inline cycledash/genotypes.py]: Confused about where we're using `run` and where we're using `vcf`; this line is a good example of both

File: cycledash/genotypes.py [hunk 3]
```diff
@@ -120,9 +120,9 @@ def get(vcf_id, query, with_stats=True):
 
         q = _add_ordering(q, g, 'String', 'contig', 'asc')
         q = _add_ordering(q, g, 'Integer', 'position', 'asc')
-
+        #print str(q.compile(compile_kwargs={"literal_binds": True}))
```
Comment (tavinathanson) [inline cycledash/genotypes.py]: Get rid of this commented line?

File: cycledash/genotypes.py (hunk 2 above)
Comment (ihodes) [inline cycledash/genotypes.py]: Yeah…this is part of a slow migration from VCF to run. I've just been changing it where I touch the code. I'll change `compareToVcfId` to `compareToRunIf`.

File: tests/python/test.py [hunk 4]
```diff
@@ -14,7 +15,7 @@ def test_underscorize():
         "CamelBI1GCasesRun1": "camel_bi1g_cases_run1"
     }
     for example, expected in camel_to_underscored_expectations.items():
-        assert underscorize(example) == expected
+        asserts.eq_(underscorize(example), expected)
```
Comment (tavinathanson) [inline tests/python/test.py]: Nit: maybe just `from nose.tools import eq_`?

File: common/helpers.py (hunk 1 above)
Comment (ihodes) [inline common/helpers.py]: Ah, I meant to call it pick. cf. http://underscorejs.org/#pick

Will fix. 

Might be worth bringing in funcy at some point?

File: cycledash/genotypes.py (hunk 3 above)
Comment (ihodes) [inline cycledash/genotypes.py]: Done

File: tests/python/test.py [hunk 5]
```diff
@@ -51,6 +52,13 @@ def test_order():
...
+    asserts.eq_(o1, [99, 22, 132])
+    asserts.eq_(o2, [{'thing': 'toga'}, {'thing': 'bathrobe'}])
+    asserts.eq_(o3, [1, -2, 2, 4, 4])
+
+
+@asserts.raises(KeyError)
+def test_pluck():
+    asserts.eq_(pluck({'a': 1, 'b': 2, 'c': 'abc'}, 'a', 'c'), {'a': 1, 'c': 'abc'})
+    asserts.eq_(pluck({'a': 1, 'b': 2, 'c': 'abc'}), {})
+    pluck({'a': 'oh no'}, 'b')  # This rasies a KeyError
```
Comment (tavinathanson) [inline tests/python/test.py]: "raises"

File: tests/python/test.py (hunk 4 above)
Comment (ihodes) [inline tests/python/test.py]: I thought about that, but I like that is still reads as "assert equal", though—what think you?

File: tests/python/test.py (hunk 5 above)
Comment (ihodes) [inline tests/python/test.py]: Yeesh, done.

File: tests/python/test_genotypes_api.py [hunk 6]
```diff
@@ -0,0 +1,206 @@
+import mock
+import nose
+import nose.tools as asserts
+import json
+
+from cycledash import db
```
Comment (tavinathanson) [inline tests/python/test_genotypes_api.py]: Do we have a story around line breaks between imports? Personally, I've been generally doing 1 group of imports for internal and (with an extra line break in between the two groups) 1 group of imports for external

Comment (ihodes) [inline tests/python/test_genotypes_api.py]: Hmm, not a great story AFAIK. I think your way is to the style guide, and I'll change this to be "external \n, cycledash-specific \n, in-current-test-directory" -- does that seem reasonable?

File: common/helpers.py (hunk 1 above)
Comment (tavinathanson) [inline common/helpers.py]: Not sure how stable/etc. `funcy` is, but I guess that's what I'd do if it's solid (or if there's something else that's solid).

File: cycledash/genotypes.py (hunk 2 above)
Comment (tavinathanson) [inline cycledash/genotypes.py]: Gotcha

File: tests/python/test.py (hunk 4 above)
Comment (tavinathanson) [inline tests/python/test.py]: Fair enough

File: tests/python/test_genotypes_api.py (hunk 6 above)
Comment (tavinathanson) [inline tests/python/test_genotypes_api.py]: :+1:

**Comment by tavinathanson** [comment]
Fly, good Fleance, fly, fly, fly (and merge)

File: cycledash/genotypes.py (hunk 2 above)
Comment (ihodes) [inline cycledash/genotypes.py]: Hmm, maybe holding off on this for now is okay? This is from the Javascript side of the application; seems to touch a lot of code… I'm also okay reverting this change if you think that'd be preferable?

Comment (tavinathanson) [inline cycledash/genotypes.py]: Fine with me

**Pull request merged**